W-14392365 Files for editing (#245)
mulesoft/docs-code-builder: modules/ROOT/pages/_editing/int-ai-developing-integrations.adoc

= Developing Integrations Using Einstein

include::partial$acb-ide-logos.adoc[tags="both-ides"]

> Use natural language prompts in Einstein for Anypoint Code Builder Generative Flows to jumpstart your integration development by generating flows for you. The engine for these generative flows are the pre-trained large language models (LLM) that exist within the Salesforce Shared Trust Boundary. These LLMs are at the core of the Generative Flows capabilities in Anypoint Code Builder. 

Reviewer: Great short description! 

"Generative Flows capabilities" sounds weird, even though I know the feature is plural.

== Trust Layer

The Einstein trust layer bridges Anypoint Platform and the LLMs, as shown in the following flow: 

> image::acb-einstein-trust-layer.png["Flow depicting the how the trust layer works with bullet points following describing the steps of the flow"]

Reviewer: ```suggestion
image::acb-einstein-trust-layer.png["Flow depicting how the trust layer works, including the steps of the flow"]
```

Reviewer: I think we need to clarify and articulate our alt text strategy. Should the alt text describe the image or should it summarize what the image is trying to convey?

> [calloutlist]

Reviewer: Yes!

Author: This one worked. But the other one never did. :-(

. The Anypoint Code Builder user crafts a prompt based on their use case and sends it to Einstein.  

> . To minimize innacurate responses, (the AI "hallucination" effect), prompts are grounded with factual MuleSoft proprietary data and user context.

Reviewer: ```suggestion
. To minimize inaccurate responses (the AI "hallucination" effect), prompts are grounded with factual MuleSoft proprietary data and user context.
```

. To ensure the data is sent safely to the external LLM, the prompt is sent to Open AI via the secure gateway. 

> . To keep your data secure, zero data is retained outside of Salesforce.  

Reviewer: zero or no? Or reword in the positive - all data remains inside Salesforce.

> . After a response is generated by the LLM, it is sent back through the secure gateway.

Reviewer: ```suggestion
. After the LLM generates a response, the content is sent back through the secure gateway.
```

Reviewer: Trying to add more active voice. Who's sending the response back? The LLM? Einstein?

. MuleSoft validates the generated output to make sure that the flow works within the MuleSoft ecosystem. 

> . The prompt and generated responses are logged for monitoring before surfacing the validated response in Anypoint Code Builder.

Reviewer: Is Einstein doing that logging and surfacing?

[IMPORTANT]

> This feature uses generative AI, which is known to include inaccurate or harmful responses. Before implementing this feature, review the output for accuracy and safety. You assume responsibility for the output when making business decisions.

Reviewer: Nice. Should this go at the beginning of the topic?
